### PR TITLE
add model and mapper for around ons content

### PR DESCRIPTION
**Pull request by AngelaLW**
### What

Add Around ONS section to homepage edit

### How to review

Get latest zebedee develop branch
Clone the feature/add-edit-ability-to-homepage-around-ons branch in the following repos:
- florence
- dp-frontend-homepage-controller
- dp-frontend-renderer

Run publishing stack locally
Create a new collection in florence CMS and attempt to edit the homepage content.
Should see a new editable list section titled 'Around ONS'
Once some content is added to the section select save and preview
In the preview window the newly added/edited Around ONS content should be displayed in the correct section

### Who can review

Anyone

### CODE REVIEW

File: homepage/homepage_updater.go [hunk 1]
```diff
@@ -83,7 +83,22 @@ func (hu *HomepageUpdater) GetHomePageUpdateFor(ctx context.Context, userAccessT
...
 
-		m := mapper.Homepage(lang, mappedMainFigures, releaseCalModelData, &mappedFeaturedContent, homepageContent.ServiceMessage)
+		var mappedAroundONS []model.Feature
+		if len(homepageContent.AroundONS) > 0 {
+			imageObjects := map[string]image.ImageDownload{}
+			for _, fc := range homepageContent.AroundONS {
+				if fc.ImageID != "" {
+					image, err := hu.clients.ImageAPI.GetDownloadVariant(ctx, userAccessToken, "", "", fc.ImageID, ImageVariant)
+					if err != nil {
+						log.Event(ctx, "error getting image download variant", log.ERROR, log.Error(err), log.Data{"around-ons-entry": fc.Title})
```
Comment (daiLlew) [inline homepage/homepage_updater.go:93]: Question:

1. Just checking that logging any error is all we want to do here? Or should we bomb out if err != nil?
2. Assuming logging is all we want to do - I think you need to add a `continue` as image will be nil - assume we should skip on to the next one in the loop?

File: mapper/mapper.go [hunk 2]
```diff
@@ -142,6 +143,20 @@ func FeaturedContent(homepageData zebedee.HomepageContent, images map[string]ima
 	return mappedFeaturesContent
 }
 
+// AroundONS takes the homepageContent as returned from the client and returns an array of featured content
+func AroundONS(homepageData zebedee.HomepageContent, images map[string]image.ImageDownload) []model.Feature {
+	var mappedAroundONS []model.Feature
+	for _, fc := range homepageData.AroundONS {
```
Comment (daiLlew) [inline mapper/mapper.go:149]: is it possible for `homepageData.AroundONS` to be nil? If so we should add a check to protect against this if necessary.

File: mapper/mapper_test.go [hunk 3]
```diff
@@ -250,6 +264,21 @@ func TestUnitMapper(t *testing.T) {
 			ImageURL:    "",
 		},
 	}
+
+	var mockedAroundONS = []model.Feature{
```
Comment (daiLlew) [inline mapper/mapper_test.go:268]: Consider adding some additional test cased to over scenarios where this is an empty struct or has some/all empty fields. I think it should be OK but tests will prove that and will safeguard us if the behaviour changes in future.

File: homepage/homepage_updater.go (hunk 1 above)
Comment (AngelaLW) [reply homepage/homepage_updater.go:93]: If the image did not exist then it should log that the image fetch has errored then set that image id as nil, so when the renderer tries to get the image further down the line it would replace the nil with a placeholder image. This functionality is the same as the current behaviour of the featured content section and is what we're trying to replicate with the Around ONS section.

File: mapper/mapper.go (hunk 2 above)
Comment (AngelaLW) [reply mapper/mapper.go]: added check for nil AroundONS property

File: mapper/mapper_test.go (hunk 3 above)
Comment (AngelaLW) [reply mapper/mapper_test.go:268]: added tests
